### ISSUE TITLE
Impossible to test mmm-mode classes in batch mode

### ISSUE THREAD

**Issue by snogge** [purcell/mmm-mode#80]
I'm trying to write `ert` tests for `bitbake-mode`, one of [bitbake-modes](https://bitbucket.org/olanilsson/bitbake-modes), that I'm the author of. 

I've run into the problem that `mmm-valid-mode` always disable `mmm-mode` when `noninteractive` is set.  And `noninteractive` is always set in batch, so it is impossible to run `ert` tests using `ert-run-tests-batch-and-exit`.

Is this just to avoid wasting cycles or is there some deeper meaning?  Is it possible to add some mechanism to allow `mmm-mode` in `noninteractive` for test purposes?

**Comment by dgutov** [comment]
> Is this just to avoid wasting cycles or is there some deeper meaning?

Probably not. But it was added in 2000, so I doubt even the author remembers.

> Is it possible to add some mechanism to allow mmm-mode in noninteractive for test purposes?

Are you calling `mmm-mode` directly? `noninteractive` is a global variable. Maybe try binding it to nil around the call?

**Comment by snogge** [comment]
No, I'm not calling `mmm-mode` directly.  It's called from my mode-function. 
Binding `noninteractive` just feels wrong, but I'll try that for now.  Would a user option for this be acceptable?

**Comment by dgutov** [comment]
> Binding noninteractive just feels wrong

When it's just in tests, why not?

**Comment by dgutov** [comment]
Anyway, let me know how it goes.
